examples/tutorials/lighting_tutorial.py

```diff
@@ -0,0 +1,226 @@
+    sensor_cfg = habitat_sim.SensorSpec()
+    sensor_cfg.resolution = [1080, 960]
+    agent_cfg = habitat_sim.agent.AgentConfiguration()
+    agent_cfg.sensor_specifications = [sensor_cfg]
+
+    return habitat_sim.Configuration(backend_cfg, [agent_cfg])
+
+
+x_axis = np.array([1, 0, 0])
+z_axis = np.array([0, 0, 1])
+y_axis = np.array([0, 1, 0])
+
+# [/setup]
+
+# This is wrapped such that it can be added to a unit test
+def main(show_imgs=True, save_imgs=False):
+    if save_imgs:
+        if not os.path.exists(output_path):
+            os.mkdir(output_path)
+
+    # [default scene lighting]
+
+    # create the simulator and render flat shaded scene
+    cfg = make_configuration()
+    sim = habitat_sim.Simulator(cfg)
+    agent_transform = place_agent(sim)
+    get_obs(sim, show_imgs, save_imgs)
+
+    # [scene swap shader]
+
+    # close the simulator and re-initialize with DEFAULT_LIGHTING_KEY:
+    sim.close()
+    cfg = make_configuration()
+    cfg.sim_cfg.scene_light_setup = habitat_sim.gfx.DEFAULT_LIGHTING_KEY
+    sim = habitat_sim.Simulator(cfg)
+    agent_transform = place_agent(sim)
+    get_obs(sim, show_imgs, save_imgs)
+
+    # create and register new light setup:
+    my_scene_lighting_setup = [
+        LightInfo(position=np.array([0.0, 2.0, 0.6]), model=LightPositionModel.GLOBAL)
+    ]
+    sim.set_light_setup(my_scene_lighting_setup, "my_scene_lighting")
+
+    # reconfigure with custom key:
+    new_cfg = make_configuration()
+    new_cfg.sim_cfg.scene_light_setup = "my_scene_lighting"
+    sim.reconfigure(new_cfg)
+    agent_transform = place_agent(sim)
+    get_obs(sim, show_imgs, save_imgs)
+
+    # [/scene]
+
+    # reset to default scene shading
+    sim.close()
+    cfg = make_configuration()
+    sim = habitat_sim.Simulator(cfg)
+    agent_transform = place_agent(sim)
+
+    # [example 2]
+
+    # load some object templates from configuration files
+    sphere_template_id = sim.load_object_configs(
+        str(os.path.join(data_path, "test_assets/objects/sphere"))
+    )[0]
+    chair_template_id = sim.load_object_configs(
+        str(os.path.join(data_path, "test_assets/objects/chair"))
+    )[0]
+
+    id_1 = sim.add_object(sphere_template_id)
+    sim.set_translation(np.array([3.2, 0.23, 0.03]), id_1)
+
+    get_obs(sim, show_imgs, save_imgs)
+
+    # [/example 2]
+
+    # [example 3]
+
+    # create a custom LightSetup
+    my_default_lighting = [
+        LightInfo(position=[2.0, 2.0, 1.0], model=LightPositionModel.CAMERA)
+    ]
+    # overwrite the default DEFAULT_LIGHTING_KEY LightSetup
+    sim.set_light_setup(my_default_lighting)
+
+    get_obs(sim, show_imgs, save_imgs)
+
+    # [/example 3]
+
+    # [example 4]
+    id_2 = sim.add_object(chair_template_id)
+    sim.set_rotation(
+        mn.Quaternion.rotation(mn.Deg(90), mn.Vector3.x_axis())
+        * mn.Quaternion.rotation(mn.Deg(-115), mn.Vector3.z_axis()),
+        id_2,
+    )
+    sim.set_translation(np.array([3.06, 0.47, 1.15]), id_2)
```

Comment:
Why not just this?

```suggestion
    sim.set_translation([3.06, 0.47, 1.15], id_2)
```

The API accepts a `mn.Vector3`, so while the above works because numpy arrays are implicitly convertible to magnum vector types, it's unnecessary. By the way, it would be good to have type annotations for the [Simulator](https://aihabitat.org/docs/habitat-sim/habitat_sim.simulator.Simulator.html) class, there's currently none (only the [SimulatorBackend](https://aihabitat.org/docs/habitat-sim/habitat_sim.sim.SimulatorBackend.html) has them because those are extracted from the native code, but that one is never used directly as far as I can see).

Same in all other cases (I see four calls to `set_translation()` with `np.array`).

Reply:
@aclegg3 And here, even though you said the math types need a general overhaul for better consistency, I nevertheless think removing the unneccessary `np.array` is a good thing to do.

Reply:
Missed this somehow, good point. My comment really was targeted at the use of numpy vs magnum types across the stack. `Simulator` is mostly using magnum types, but `AgentState`, for example, is using numpy.